File: app/src/main/java/com/tim/common/ICallback.java
package com.tim.common;

/**
 * ICallback
 *
 * @author Tell.Tim
 * @date 2019/12/24 19:44
 */
public interface ICallback<S, F>{
    void onSuccess(S s);
    void onFail(F f);
    void onError(Throwable throwable);
}

File: app/src/main/java/com/tim/common/ICallback.kt
package com.tim.common

/**
 * ICallback
 *
 * @author Tell.Tim
 * @date 2019/12/24 19:44
 */
interface ICallback<S, F> {
    fun onSuccess(s: S)
    fun onFail(f: F)
    fun onError(throwable: Throwable)
}
